__app__/crop/structure.py

```diff
@@ -384,6 +386,55 @@ class ReadingsAranetTRHClass(BASE):
     __table_args__ = (UniqueConstraint("sensor_id", "timestamp"),)
 
 
+class ReadingsAranetCO2Class(BASE):
+    """
+    Base class for the Aranet CO2  GU sensor readings
+    """
+    __tablename__ = ARANET_CO2_TABLE_NAME
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    sensor_id = Column(
+        Integer,
+        ForeignKey("{}.{}".format(SENSOR_TABLE_NAME, ID_COL_NAME)),
+        nullable=False,
+    )
+
+    timestamp = Column(DateTime, nullable=False)
+    co2 = Column(Float, nullable=False) # units of parts-per-million
+
+    time_created = Column(DateTime(), server_default=func.now())
+    time_updated = Column(DateTime(), onupdate=func.now())
+
+    # arguments
+    __table_args__ = (UniqueConstraint("sensor_id", "timestamp"),)
+
+
+class ReadingsAranetAirVelocityClass(BASE):
+    """
+    Base class for the Aranet air velocity GU sensor readings.
+    Record both the raw current from the sensor, and the corresponding calibrated
+    air velocity.
+    """
+    __tablename__ = ARANET_AIRVELOCITY_TABLE_NAME
+    # columns
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    sensor_id = Column(
+        Integer,
+        ForeignKey("{}.{}".format(SENSOR_TABLE_NAME, ID_COL_NAME)),
+        nullable=False,
+    )
+
+    timestamp = Column(DateTime, nullable=False)
+    current = Column(Float, nullable=False) # raw current, in Amps
```

Comment:
Might we want to leave current be nullable? It seems like an optional bit of data to me, e.g. I could imagine a slightly different air velocity sensor that didn't report it.

Reply:
yep, fair enough, changed this to nullable.